File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 9/16/2025, 12:52:59 AM
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        i = 0 

        while(i<len(nums)):

            if nums.count(nums[i]) > len(nums)//2:
                return nums[i]

            else:
                i = next((j for j in range(i, len(nums)) if nums[j] != nums[i]), len(nums))


        

File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 2/13/2026, 12:57:44 PM
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
        int c = 0, el = nums[0];
        for(int i = 0; i<nums.size(); i++)
        {
            if( nums[i] == el) c++;
            else c--;

            if(c == 0) el = nums[i+1];
        }

        return el;
    }
};